### PR TITLE
[WIP] Subscriptions instead of License keys

### PR DESCRIPTION
**Pull request by rooftopcellist**
##### SUMMARY

Overhaul of licensing in favor of using RH subscriptions model.  

subscriptions-ui-flow

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

 - API
 - UI
 - Installer

### CODE REVIEW

File: awx/ui/client/src/license/license.controller.js [hunk 1]
```diff
@@ -44,11 +44,11 @@ export default
...
 
             if ($rootScope.licenseMissing) {
-                $scope.title = $rootScope.BRAND_NAME + i18n._(" License");
+                $scope.title = $rootScope.BRAND_NAME + i18n._(" Subscription");
             } else {
-                $scope.title = i18n._("License Management");
+                $scope.title = i18n._("Subscription Management");
             }
-
+            // What if I POST the file as a field in the License data JSON?
```
Comment (mabashian) [inline awx/ui/client/src/license/license.controller.js]: Still need this?

File: awx/ui_next/src/variables.js [hunk 2]
```diff
@@ -1,2 +1,2 @@
-export const BrandName = 'AWX';
```
Comment (mabashian) [inline awx/ui_next/src/variables.js]: Probably don't want to commit this change to devel

File: awx/ui/client/src/license/license.partial.html [hunk 3]
```diff
@@ -79,7 +73,7 @@
 		<div class="card at-Panel">
 			<div class="List-titleText">{{title}}</div>
 			<div class="License-body">
-				<div class="License-helperText License-introText" ng-if="licenseMissing" translate>Welcome to Ansible Tower! Please complete the steps below to acquire a license.</div>
+				<div class="License-helperText License-introText" ng-if="licenseMissing" translate>Welcome to Ansible Tower! Please complete the steps below to acquire a Subscription.</div>
```
Comment (jlmitch5) [inline awx/ui/client/src/license/license.partial.html]: sentence casing

File: awx/ui/client/src/license/license.partial.html [hunk 4]
```diff
@@ -88,29 +82,29 @@
 									1
 								</span>
 								<span class="License-helperText">
-									<translate>Please click the button below to visit Ansible's website to get a Tower license key.</translate>
+									<translate>Please click the button below to visit Ansible's website to get a Tower Subscription.</translate>
```
Comment (jlmitch5) [inline awx/ui/client/src/license/license.partial.html]: sentence casing

File: awx/ui/client/src/license/license.partial.html [hunk 5]
```diff
@@ -125,12 +119,12 @@
 						<div class="d-block w-100">
 							<div class="AddPermissions-directions">
 								<span class="License-helperText">
-									<translate>Provide your Red Hat customer credentials and you can choose from a list of your available licenses.  The credentials you use will be stored for future use in retrieving renewal or expanded licenses. You can update or remove them in SETTINGS > SYSTEM.</translate>
+									<translate>Provide your Red Hat or Satellite credentials below and you can choose from a list of your available Subscriptions.  If you are a Satellite user please specify the organization ID you wish to associate this Tower Installation to.  The credentials you use will be stored for future use in retrieving renewal or expanded Subscriptions.</translate>
```
Comment (jlmitch5) [inline awx/ui/client/src/license/license.partial.html]: sentence casing

**Review by jlmitch5 — COMMENTED** [review]
Subscription should not be capitalized in these places

File: awx/main/validators.py [hunk 6]
```diff
@@ -17,6 +17,12 @@
 from awx.main.utils import parse_yaml_or_json
 
 
+def validate_entitlement_cert(data, min_keys=0, max_keys=None, min_certs=0, max_certs=None):
+    # TODO: replace with more actual robust logic here to allow for multiple certificates in one cert file (because this is how entitlements do)
```
Comment (shanemcd) [inline awx/main/validators.py:21]: This comment exists to make sure we dont forget about this.

File: requirements/requirements_git.txt [hunk 7]
```diff
@@ -1 +1,2 @@
 git+https://github.com/ansible/system-certifi.git@devel#egg=certifi
+git+https://github.com/candlepin/python-rhsm.git@python3-backport#egg=rhsm
```
Comment (shanemcd) [inline requirements/requirements_git.txt]: We should fork this under the Ansible org.

**Review by shanemcd — CHANGES_REQUESTED** [review]
Overall looks pretty good. Just added a couple small notes on things we dont want to forget.

File: awx/main/utils/licensing.py [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+    )
+
+    def __init__(self, **kwargs):
+        self._attrs = dict(
+            instance_count=0,
+            license_date=0,
+            license_type='UNLICENSED',
+        )
+        if not kwargs:
+            license_setting = Setting.objects.filter(key='LICENSE').first()
```
Comment (ryanpetrello) [inline awx/main/utils/licensing.py]: Shouldn't this just be `settings.LICENSE` ?  Is this something you were experimenting with in relation to https://github.com/ansible/awx/pull/8209/commits/5ef2c94f63cc1232b35fa4aed298a8ed9e4ffad8 ?

**Comment by shanemcd** [comment]
recheck

**Pull request closed without merging**